src/librustc_typeck/check/method/suggest.rs

```diff
@@ -518,7 +518,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     }
                 }
 
-                if let Some(lev_candidate) = lev_candidate {
+                fn is_str_ref(ty: Ty<'_>) -> bool {
+                    match ty.sty {
+                        ty::Str => true,
+                        ty::Ref(_, ty, _) => is_str_ref(&ty),
+                        _ => false,
+                    }
+                }
+                if item_name.as_str() == "as_str" && is_str_ref(&actual) {
```

Comment:
This should use the diagnostics item infra I think. cc @oli-obk

Reply:
Not quite. There is no item here to slap the rustc_diagnostic_item attribute on. This lint suggests removing a call to a nonexistant function.

Reply:
Ah, good point, yes.